Add initial documentation for Multi-Screen Window Placement (#18)
GoogleChromeLabs/puppy-content: content/en-US/api/getScreens.md

---
recipe: api-method

> title: 'getScreens'

Reviewer: ```suggestion
title: 'getScreens()'
```

Reviewer: I've added this to the template.

mdn_url: /en-US/docs/Web/API/Window/getScreens
specifications: https://webscreens.github.io/window-placement/#dom-window-getscreens
browser_compatibility: api.Window.getScreens
---

## Description

The `getScreens()` method of the `Window` interface returns information about each connected screen. The method may prompt the user for permission.

## Syntax

`Window.getScreens().then(...)`

### Parameters

None.

### Return value

> Returns a {{jsxref('Promise')}} that resolves with a sequence of ScreenInfo objects.

Reviewer: ```suggestion
Returns a Promise that resolves with a sequence of ScreenInfo objects.
```

Reviewer: This is an MDN macro that I don't have a way to resolve right night. I'm removing these as well.

Author: fix coming

> Each ScreenInfo object has the following properties:

Reviewer: ```suggestion
Each `ScreenInfo` object has the following properties:
```

Author: Obsolete

> <dl>

Reviewer: Please close your tags.

Author: Obsolete. (Also... heh, does MDN use XHTML or something? Closing tags is for weenies.)

  <dt>`availWidth`
  <dd>The width of the web-exposed available screen area.
  <dt>`availHeight`
  <dd>The height of the web-exposed available screen area.
  <dt>`width`
  <dd>The width of the web-exposed screen area.
  <dt>`height`
  <dd>The height of the web-exposed screen area.

>   <dt>`colorDepth`

Reviewer: There's a definition missing. Just copy the `pixelDepth`. It's an MDN thing.

Author: Obsolete

  <dt>`pixelDepth`

>   <dd>The number of bits allocated to colors for a pixel in the output device, excluding the alpha channel. The `colorDepth` and `pixelDepth` values are the same, for compatibility reasons.

Reviewer: What does it mean that these are the same for compatibility reasons?

  <dt>`orientationType`

>   <dd>One of "portrait-primary",  "portrait-secondary",  "landscape-primary",  "landscape-secondary".

Reviewer: ```suggestion
  <dd>One of "portrait-primary",  "portrait-secondary",  "landscape-primary",  or "landscape-secondary".</dd>
```
